src/build/utils/kdl.ts

```diff
@@ -0,0 +1,153 @@
+import { parse } from "kdljs";
+import { Enum, Event, Method, Property } from "../types";
+import { readdir, readFile } from "fs/promises";
+import { merge } from "../helpers.js";
+
+interface MethodDescriptor {
+  [x: string]: Omit<Method, "signature">;
+}
+
+interface EnumDescriptor {
+  [x: string]: Enum;
+}
+interface PropertyDescriptor {
+  [x: string]: Omit<Property, "type">;
+}
+
+interface InterfaceMixin {
+  [x: string]: {
+    events?: { event: Event[] };
+    methods?: { method: MethodDescriptor };
+  };
+}
+
+/**
+ * Converts KDL text describing interface-mixins into JSON format.
+ * Uses kdljs v3.
+ */
+export function parseKDL(kdlText: string) {
+  const { output, errors } = parse(kdlText);
+
+  if (errors.length) {
+    throw new Error(
+      `KDL parse errors:\n${errors.map((e) => e.message).join("\n")}`,
+    );
+  }
+
+  const nodes = output!;
+  const enums: EnumDescriptor = {};
+  const interfaces: InterfaceMixin = {};
+
+  for (const node of nodes) {
+    if (node.name === "enum") {
+      // Handle enum
+      const enumName = node.values?.[0]?.toString() ?? "";
+      let name = enumName;
+      const values: string[] = [];
+
+      for (const child of node.children ?? []) {
+        if (
+          child.name === "value" &&
+          child.values &&
+          child.values[0] !== undefined
+        ) {
+          values.push(child.values[0]!.toString());
+        } else if (child.name === "name") {
+          name = child.values?.[0]?.toString() ?? enumName;
+        }
```

Comment:
```suggestion
        if (child.name !== "value" || typeof child.values[0] !== "string") {
          throw new Error("enum values should be in the form of `value {name}`");
        }
        values.push(child.values[0]);
```

1. We should throw when the items are not in the expected form.
2. I see why `child.name === "name"` branch exists... but let's exclude that here, given that 1) the added KDL file doesn't use this branch yet, and 2) I think it should be a property instead of a child node (maybe `override-name=(newname)`).